### ISSUE TITLE
Thunderbolt egpu doesn't work in catalina

### ISSUE THREAD

**Issue by PuPuppru** [jaromeyer/XPS9570-Catalina#16]
As the titile the gpu(vega56) can't boot into osx but it works in windows,any knows how to deal with it?

**Comment by jaromeyer** [comment]
im not using thunderbolt myself, but have you tried setting Thunderbolt Adapter Configuration to "all on, no security" and selecting BIOS enumeration?

**Comment by PuPuppru** [comment]
> im not using thunderbolt myself, but have you tried setting Thunderbolt Adapter Configuration to "all on, no security" and selecting BIOS enumeration?

Yes,I did that,and with LuletterSoul's efi the system can recognize the egpu,although the second monitor still doesn't work

**Comment by PuPuppru** [comment]
> im not using thunderbolt myself, but have you tried setting Thunderbolt Adapter Configuration to "all on, no security" and selecting BIOS enumeration?

I think maybe u can try this https://github.com/LuletterSoul/Dell-XPS-15-9570-macOS-Mojave/issues/4#issuecomment-506429102

**Comment by PuPuppru** [comment]
> > im not using thunderbolt myself, but have you tried setting Thunderbolt Adapter Configuration to "all on, no security" and selecting BIOS enumeration?
> 
> I think maybe u can try this [LuletterSoul/Dell-XPS-15-9570-macOS-Mojave#4 (comment)](https://github.com/LuletterSoul/Dell-XPS-15-9570-macOS-Mojave/issues/4#issuecomment-506429102)

I add those ssdt into the efi ,the system can recognise the thunderbolt port,but it still can't find egpu ports.

**Comment by frbuccoliero** [comment]
> > > im not using thunderbolt myself, but have you tried setting Thunderbolt Adapter Configuration to "all on, no security" and selecting BIOS enumeration?
> > 
> > 
> > I think maybe u can try this [LuletterSoul/Dell-XPS-15-9570-macOS-Mojave#4 (comment)](https://github.com/LuletterSoul/Dell-XPS-15-9570-macOS-Mojave/issues/4#issuecomment-506429102)
> 
> I add those ssdt into the efi ,the system can recognise the thunderbolt port,but it still can't find egpu ports.

With that patch you can hotplug thunderbolt? Wow 😮 
Let's get this eGPU working then. Are you sure isn't there any additional driver provided by the eGPU manufacturer to install?

**Comment by PuPuppru** [comment]
> > > > im not using thunderbolt myself, but have you tried setting Thunderbolt Adapter Configuration to "all on, no security" and selecting BIOS enumeration?
> > > 
> > > 
> > > I think maybe u can try this [LuletterSoul/Dell-XPS-15-9570-macOS-Mojave#4 (comment)](https://github.com/LuletterSoul/Dell-XPS-15-9570-macOS-Mojave/issues/4#issuecomment-506429102)
> > 
> > 
> > I add those ssdt into the efi ,the system can recognise the thunderbolt port,but it still can't find egpu ports.
> 
> With that patch you can hotplug thunderbolt? Wow 
> Let's get this eGPU working then. Are you sure isn't there any additional driver provided by the eGPU manufacturer to install?

ummm，in fact , it isn't completely hotplug, need to connect the devices before booting, and my eGPU  is vega56 so I don't think I need to install any driver

**Comment by theeno513** [comment]
have you solve the problem?

**Comment by theeno513** [comment]
I plan to buy a mantiz and rx580, but I'm wondering if it can work.

**Comment by PuPuppru** [comment]
> I plan to buy a mantiz and rx580, but I'm wondering if it can work.

Sorry,still no idea to fix it

**Comment by jaromeyer** [comment]
sorry guys but I don't have any thunderbolt devices myself, so I cant look into this issue for the time being
